@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -846,12 +846,15 @@ export class VisibleTileSet {
      * the {@link DataSource} with the given name.
      *
      * @param dataSourceName - The name of the {@link DataSource}.
+     * @param filter Optional tile filter
      */
-    clearTileCache(dataSource?: DataSource) {
+    clearTileCache(dataSource?: DataSource, filter?: (tile: Tile) => boolean) {
         if (dataSource !== undefined) {
             this.m_dataSourceCache.evictSelected((tile: Tile, _) => {
-                return tile.dataSource === dataSource;
+                return tile.dataSource === dataSource && (filter ? filter(tile) : true);
```

Comment:
Are we now allowing truthy values?

Reply:
Not if you are sharp enough to see it, fixed!